src/arch/x86/idle.S

```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright 2025, UNSW
+ *
+ * SPDX-License-Identifier: GPL-2.0-only
+ */
+
+#include <machine/assembler.h>
+
+.section .text, "ax"
+.code64
```

Comment:
This needs to be `.code32` if you want it to be common between 32-bit and 64-bit.